include/at_chops/aes_ctr.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#ifdef BUILD_MBEDTLS
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <string.h>
+
+typedef struct {
+    unsigned char *res; // result of the encryption/decryption
+    size_t reslen; // length of the result written
+} AESResult;
+
+/**
+ * @brief AES 256 CTR encrypt plaintext
+ * 
+ * @param key_base64 the base64 encoded AES 256 key (e.g. "1DPU9OP3CYvamnVBMwGgL7fm8yB1klAap0Uc5Z9R79g=")
+ * @param plaintext the plain text to encryt, must be null terminated `\0`
+ * @return AESResult* the result of the encryption
+ */
+AESResult *atchops_aes256ctr_encrypt(const char *key_base64, const unsigned char *plaintext);
```

Comment:
Encrypt and Decrypt are both missing a way of handling errors.

Typically this is done with an int return value, with zero for success and errors representing other values. The result would instead be passed into the function as a pointer, which also allows the end developer to choose their own buffer size.

An alternative which could be done here is to add an int to your AESResult, which would indicate success / error

Reply:
Also, there are still declarations in at_chops.h as well, please remove the ones which aren't being used (I assume the other ones aren't)

Reply:
Done in https://github.com/atsign-foundation/at_c/pull/7/commits/332615e1651372707d046cfd109c4ce6d893aa19